### PR TITLE
translation(js): /Web/JavaScript/Reference/Operators/typeof

### PR DESCRIPTION
**Pull request by AdriandeCita**
Original content: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/typeof

### CODE REVIEW

**Comment by AdriandeCita** [comment]
Оновлено, нові зміни: [abcebf4](https://github.com/mdn/content/commit/abcebf471d56ef12239e2565f26d952e8a8cab2e), [df6360a](https://github.com/mdn/content/commit/df6360a23ca7c231b4bf1340b028440344bbc159), [4e55ef8](https://github.com/mdn/content/commit/4e55ef8d3618f047997d8670fce06b838de6a17b)

File: files/uk/web/javascript/reference/operators/typeof/index.md [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+typeof Number('1') === 'number';      // Конструктор Number намагається перетворити значення на числа
+typeof Number('shoe') === 'number';   // навіть ті, які неможливо привести до числа
+
+typeof 42n === 'bigint';
+
+// Рядки
+typeof '' === 'string';
+typeof 'bla' === 'string';
+typeof `шаблонний літерал` === 'string';
+typeof '1' === 'string'; // зауважте, що число всерединні рядка все ж має рядковий тип
```
Comment (undead404) [inline files/uk/web/javascript/reference/operators/typeof/index.md:76]:
```suggestion
typeof '1' === 'string'; // зауважте, що число всередині рядка все ж має рядковий тип
```

File: files/uk/web/javascript/reference/operators/typeof/index.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+### `typeof null`
+
+```js
+// Ця рівність зберігається з часів створення JavaScript
+typeof null === 'object';
+```
+
+В першій реалізації JavaScript, його значення позначалися як тег типу та власне значення. Тег типу для об'єктів був `0`. `null` позначався як нульовим вказівником (на більшості платформ це `0x00`). Як наслідок, `null` мав тег типу `0`, тому `typeof` повертає значення `"object"`. ([посилання (англ.)](https://2ality.com/2013/10/typeof-null.html))
+
+Було запропоновано виправлення для ECMAScript (як "opt-in" — можливість, яку можна було ввімкнути), проте [його відхилили](https://web.archive.org/web/20160331031419/http://wiki.ecmascript.org:80/doku.php?id=harmony:typeof_null). Результатом його прийняття була б рівність `typeof null === 'null'`.
```
Comment (undead404) [inline files/uk/web/javascript/reference/operators/typeof/index.md:126]:
```suggestion
Було запропоновано виправлення для ECMAScript (як "opt-in" — можливість, яку можна було б увімкнути), проте [його відхилили](https://web.archive.org/web/20160331031419/http://wiki.ecmascript.org:80/doku.php?id=harmony:typeof_null). Результатом його прийняття була б рівність `typeof null === 'null'`.
```

File: files/uk/web/javascript/reference/operators/typeof/index.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+### Винятки
+
+Всі актуальні браузери дають змогу доступитися до нестандартного об'єкта системного об'єкта [`document.all`](/uk/docs/Web/API/Document/all)
+з типом `undefined`.
+
+```js
+typeof document.all === 'undefined';
+```
+
+Хоча специфікація дає змогу мати користувацькі теги типу для нестандартних екзотичних об'єктів, вона вимагає, щоб ці теги відрізнялися від раніше заданих. Випадок з `document.all`, що має тип `'undefined'` класифіковано у вебстандартах як "навмисне порушення" первісного стандарту ECMA JavaScript.
```
Comment (undead404) [inline files/uk/web/javascript/reference/operators/typeof/index.md:189]:
```suggestion
Хоча специфікація дає змогу мати користувацькі теги типу для нестандартних екзотичних об'єктів, вона вимагає, щоб ці теги відрізнялися від раніше заданих. Випадок з `document.all`, що має тип `'undefined'`, класифіковано у вебстандартах як "навмисне порушення" первісного стандарту ECMA JavaScript.
```

File: files/uk/web/javascript/reference/operators/typeof/index.md [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+
+### Застосування в реальному світі
+
+Оператор `typeof` дуже корисний, проте він не настільки універсальний, наскільки це може бути потрібно. Наприклад, `typeof([])` дорівнює `'object'`, так само як `typeof(new Date())`, `typeof(/abc/)` та ін.
+
+Для більшої специфічности під час перевірки типів, можна зробити таку обгортку над `typeof` для використання в робочому коді (за умови, що `obj` існує):
```
Comment (undead404) [inline files/uk/web/javascript/reference/operators/typeof/index.md:195]:
```suggestion
Для більшої специфічности під час перевірки типів можна зробити таку обгортку над `typeof` для використання в робочому коді (за умови, що `obj` існує):
```

**Review by undead404 — CHANGES_REQUESTED** [review]

**Review by undead404 — APPROVED** [review]

**Pull request merged**